Engine_Revit_UI/Modify/SetParameter.cs

```diff
@@ -133,9 +133,33 @@ public static bool SetParameter(this Element element, BuiltInParameter builtInPa
 
         public static bool SetParameter(this Parameter parameter, object value, Document document = null)
         {
+            // Skip null and read-only parameters
             if (parameter == null || parameter.IsReadOnly)
                 return false;
 
+            // Workset parameters
+            // Filter for worksets
+            if (parameter.Id.IntegerValue == (int)BuiltInParameter.ELEM_PARTITION_PARAM)
+            {
+                // Find an existing workset with a specified name if it exists
+                string worksetName = value as string;
+                Workset workset = document.Workset(worksetName);
+
+                // Set the "Workset" parameter to the specified existing workset
+                // Ensure the Query method hasn't returned a null, which can happen if the document is not workshared or the workset name is empty
+                if (workset != null)
+                {
+                    // Set the parameter to a workset with the specified name if it exists
+                    return parameter.Set(workset.Id.IntegerValue);
+                }
+                else
+                {
+                    // A workset with the specified name doesn't exist
+                    BH.Engine.Reflection.Compute.RecordError("Cannot set the Workset parameter because a workset with the specified name doesn't exist.");
```

Comment:
I would change it to a warning - errors are reserved to cases when an issue occurs that results in wrong/no result. This is just a single parameter not being set for a non-breaking reason, so warning sounds more suitable I think.

```suggestion
                    BH.Engine.Reflection.Compute.RecordWarning("Cannot set the Workset parameter because a workset with the specified name doesn't exist.");
```